netmanager/src/views/components/DataDisplay/Devices.js

```diff
@@ -184,71 +188,171 @@ const MenuProps = {
   };
 
   const [hasError, setHasError] = useState(false); 
-
-
   //maintenance log parameters
+  const maintenanceOptions = [
+    "Dust blowing and sensor cleaning",
+    "Site update check",
+    "Device equipment check",
+    "Power circuitry and components works",
+    "GPS module works/replacement",
+    "GSM module works/replacement",
+    "Battery works/replacement",
+    "Power supply works/replacement",
+    "Antenna works/replacement",
+    "Mounts replacement",
+    "Software checks/re-installation",
+    "PCB works/replacement",
+    "Temp/humidity sensor works/replacement",
+    "Air quality sensor(s) works/replacement"
+  ];
+
   const [deviceName, setDeviceName] = useState('');
-  const [maintenanceDescription, setMaintenanceDescription] = useState('');
+  const [maintenanceType, setMaintenanceType] = useState('');
+  const handleMaintenanceTypeChange = type =>{
+    setMaintenanceType(type.target.value);
+    if (type.target.value == 'Preventive Maintenance'){
+      setMaintenanceDescription(["Dust blowing and sensor cleaning","Site update check","Device equipment check",])
+    }
+    else{
+      setMaintenanceDescription([]);
+    }
+  }
+  //const [maintenanceDescription, setMaintenanceDescription] = useState('');
+  const [maintenanceDescription, setMaintenanceDescription] = useState([]);
   const handleMaintenanceDescriptionChange = description => {
     setMaintenanceDescription(description.target.value);
   } 
+
   const [selectedDate, setSelectedDate] = useState(new Date());
   const handleDateChange = (date) => {
     setSelectedDate(date);
   };
   //sensor parameters
+  const quantityOptions = ["PM 1(µg/m3)", "PM 2.5(µg/m3)", "PM 10(µg/m3)", "External Temperature(\xB0C)", 
+  "External Temperature(\xB0F)", "External Humidity(%)", "Internal Temperature(\xB0C)", "Internal Humidity(%)", 
+  "Battery Voltage(V)", "GPS"];
+
+  const convertQuantityOptions= (myArray) => {
+    let newArray = [];
```

Comment:
I noticed similar code else where (`netmanager/src/views/components/DataDisplay/DeviceView.js`), could we explore code reusability?

Reply:
It can be looked into in a future PR